File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/MotoboyOperationalStatusTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class MotoboyOperationalStatusTest {
  @Test
  fun `shows paused status when courier is unavailable`() {
    val status = motoboyOperationalStatus(available = false, trackingEnabled = true)

    assertEquals("Corridas paradas", status.title)
  }

  @Test
  fun `shows available status when courier can accept deliveries`() {
    val status = motoboyOperationalStatus(available = true, trackingEnabled = false)

    assertEquals("Recebendo corridas", status.title)
  }

  @Test
  fun `shows automatic GPS status without promising background location`() {
    val status = motoboyOperationalStatus(available = true, trackingEnabled = true)

    assertEquals("Recebendo corridas", status.title)
    assertEquals(
      "Novas entregas podem aparecer aqui. Sua localizacao ajuda a loja a acompanhar o atendimento.",
      status.text,
    )
  }
}

File: apps/motoboy-pwa/src/motoboyOperationalStatus.test.ts
import assert from "node:assert/strict";
import test from "node:test";
import { motoboyOperationalStatus } from "./motoboyOperationalStatus";

test("shows paused status when courier is unavailable", () => {
  const status = motoboyOperationalStatus({ available: false, trackingEnabled: true });

  assert.equal(status.tone, "paused");
  assert.equal(status.title, "Corridas paradas");
});

test("shows available status when courier can accept deliveries", () => {
  const status = motoboyOperationalStatus({ available: true, trackingEnabled: false });

  assert.equal(status.tone, "ready");
  assert.equal(status.title, "Recebendo corridas");
});

test("shows automatic GPS status when open-app tracking is active", () => {
  const status = motoboyOperationalStatus({ available: true, trackingEnabled: true });

  assert.equal(status.tone, "active");
  assert.equal(status.title, "Recebendo corridas");
});
